### ISSUE TITLE
Spurious unread counts for Gmail; only solution is to delete the mail

### ISSUE THREAD

**Issue by TheBloke** [wavebox/waveboxapp#746]
* Wavebox Version: 3.14.10beta (but has affected earlier versions also)
* Operating System & Version: Mac OS 10.13.6  (has happened on 10.13.4 and 10.13.5 also)
* Account type (if applicable): Gmail
* I have a pro license.

**Expected & actual behavior**
On three occasions over the last few weeks I have had Wavebox suddenly start reporting that I have one item of unread mail, even though all mail in Gmail is definitely read.  The item of mail Wavebox shows as unread is not a new mail, but appears to suddenly be regarded as unread 1-2 days after it first arrived.  

In one case, it was an item of _sent_ mail that Wavebox said was unread - which should never happen.

In all three cases:
* An old mail is suddenly regarded as unread by Wavebox with no obvious interaction with that mail, or with Wavebox.  The issue does not seem correlated with stop/starts of Wavebox, as I keep Wavebox running 24/7 (my workstation is never put to sleep.)
    * It may be triggered by new mail coming in, but I can't be certain.  I only notice the problem when I've marked all mail as read and notice Wavebox still shows a count of 1 unread.
* The email gets regarded as unread a while after it first arrived and was first read - Wavebox suddenly thinks it is 'unread' some hours or days after I last interacted with it.
    * In this most recent case, the affected mail arrived and was marked read 45 hours before Wavebox reported it as being 'unread' again.
    * On the occasion before this, the affected mail was approx 24 hours old.
* That mail, and all other, shows as read in Wavebox and in any other GMail client;
* Therefore the only way I can work out which mail is responsible for the '1 unread' count is to use Wavebox's menu bar, which shows unread items;
* Marking that mail, or all mail, as unread + read again makes no difference;
* Quitting and restarting Wavebox does not help;
* Reload, Resync and Reauthenticate do not help;
* Archiving the message out of the Inbox also made no difference, nor moving it back again;
* Receiving new email and marking it read does not affect the spurious unread message: in effect the count is now always 1 + _number of unread messages_, rather than starting at 0.

The only solution I have found for this problem is to delete the mail that Wavebox believes is 'unread'.  If I don't do that, I will have a permanent, spurious unread count forever.  The only thing I haven't tried is deleting the whole account from Wavebox and re-adding it.  I imagine that would fix it also, but is a hassle in terms of the re-configuration required.

As mentioned above, on a previous occurrence of this issue the mail falsely reported as unread was actually a _sent_ mail, which makes even less sense as this should never be marked 'unread'.  To me this suggests that the problem is nothing to do with mail actually being read/unread, but perhaps some kind of indexing problem, which can affect any kind of mail?

**Steps to reproduce**
Unknown.

**Is the bug persistent or intermittent?**
Has happened a total of three times over approximately 8 weeks: 
* today (1st August)
* 4th July (exactly four weeks ago)
* and some weeks before that. I can't remember exactly when the first occasion was.

Once the bug appears it is permanent, save for the undesirable workaround of having to delete the affected mail.  This means I don't have the mails that caused the issue the first two times.  But I will keep the mail affected by this third occurrence, in case it can aid in debugging.

**Screenshots**

**Menu bar icon shows which mail is being considered as 'unread':**
![screenshot_2018-08-01 12 25 20_xk8fjb](https://user-images.githubusercontent.com/784313/43519119-1a84854e-9586-11e8-9ddb-1762fb27dab5.png)

**Main UI showing unread count of one, and showing that that message is in fact read:**
![screenshot_2018-08-01 12 44 19_t4hpjh](https://user-images.githubusercontent.com/784313/43519785-abbd9abc-9588-11e8-8caa-7ebe64a11e84.png)

**Comment by Thomas101** [comment]
Hi thanks for reporting. We saw something similar to this when the new Gmail first launched, the API would start reporting some items as unread when they weren't.

A little bit of investigation revealed that emails with Google Inbox assists & new Gmail nudges on them seemed to trigger as unread. We reported this to Google and it did seem to subside after a while but it could have returned again. From what you've described (e.g. the email appearing as unread again after a few days) it sounds like it could be this, but it would be good to confirm it.

Do you ever use Google Inbox or the new Gmail on mobile or the web? If you do, it might be worth checking if the emails have been highlighted for replying etc...
<img src="https://www.androidpolice.com/wp-content/uploads/2017/09/nexus2cee_DJOqG0nXkAENOyh.png" width="45%" />
<img src="https://i.gadgets360cdn.com/large/gmail_main_1526376386474.jpg?output-quality=80" width="45%" />

If you're not using the new Gmail or Google Inbox but the assists/nudges have triggered, then it may be worth turning them off in settings to see if this stops the count reporting incorrectly.

We also added some experimental unread modes which use a different method to grab the unread count to try and alleviate the original issue. These have had mixed results, so they may be more resilient for you or you may find that it's worse, but it may be worth trying. To change it, go to your account settings and find the unread mode. Most of them have an equivalent experimental mode, change it to one of those and see if it helps.

**Comment by TheBloke** [comment]
Thanks for the details, @Thomas101 

I have not been using Inbox or the new Gmail anywhere.  I did very briefly switch to New Gmail inside Wavebox, but changed back within 5 minutes as I didn't like the new format.

I tried logging in on the web and switching to New Gmail, and no nudges were visible against the affected mail.  I then tried disabling Nudges in the settings, and that made no difference.

But I've now changed Wavebox to the Experimental version of my Unread method (Unread Messages in Primary Category) and that immediately set my unread count to the correct value, 0.  I've since received a couple of emails and the count went correctly to 1, then back to 0 when they were read.

So that seems to have solved the problem for me, at least for now.  I'll keep an eye on it and report back if I get any further problems using the new Experimental method.

Thanks again.
